common/logisticspipes/utils/string/StringUtils.java

```diff
@@ -110,10 +110,12 @@ public static String getFormatedStackSize(long stackSize, boolean forceDisplayNu
 			s = stackSize + "";
 		} else if (stackSize < 100000) {
 			s = stackSize / 1000 + "K";
-		} else if (stackSize < 1000000) {
-			s = "0M" + stackSize / 100000;
-		} else {
+		} else if (stackSize < 10000000) {
+			s = (int)stackSize / 1000000 + "M" + (stackSize - ((int)stackSize / 1000000) * 1000000) / 100000;
```

Comment:
I don't know why you added the casts to int. The part after M (also with modification below for G) can be simplified to `(stackSize % 1000000) / 100000`

Reply:
I added the casts to int because it rounds down and returns an int, instead of using `Math.floor()` and `Integer.parseInt()` to accomplish the same.
And you are completely right about the `(stackSize % 1000000) / 100000` part it is better to make it that way I completely forgot about that.

Reply:
it doesn't use `Math.floor()` here. neither parts are of type float or double :)

Reply:
I know it doesn't, I needed to use it to isolate the highest value digit